src/__tests__/App.test.tsx

```diff
@@ -0,0 +1,115 @@
+  it("should not be possible to add a flashcard when front card value is empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+  });
+
+  it("should not be possible to add a flashcard when back card value is empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+
+    newCardInput = await screen.findByRole("textbox");
+
+    fireEvent.change(newCardInput, { target: { value: "" } });
+    fireEvent.click(saveCardButton);
+    expect(saveCardButton).toBeInTheDocument();
+  });
+
+  it("should be possible to add a flashcard when front and back card value is not empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+
+    newCardInput = await screen.findByRole("textbox");
+
+    fireEvent.change(newCardInput, { target: { value: "" } });
+    fireEvent.click(saveCardButton);
+    expect(saveCardButton).toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is back" } });
+    fireEvent.click(saveCardButton);
```

Comment:
**issue:** You should check if added card is visible with given values in cards list after clicking on Save